cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -260,18 +260,32 @@ class fixed_point {
   fixed_point() : _value{0}, _scale{scale_type{0}} {}
 
   /**
-   * @brief Explicit conversion operator
+   * @brief Explicit conversion operator for casting to floating point types
    *
-   * @tparam U The type that is being explicitly converted to (integral or floating)
+   * @tparam U The floating point type that is being explicitly converted to
    * @return The `fixed_point` number in base 10 (aka human readable format)
    */
   template <typename U,
-            typename cuda::std::enable_if_t<is_supported_construction_value_type<U>()>* = nullptr>
+            typename cuda::std::enable_if_t<std::is_floating_point<U>::value>* = nullptr>
   CUDA_HOST_DEVICE_CALLABLE explicit constexpr operator U() const
   {
     return detail::shift<Rep, Rad>(static_cast<U>(_value), detail::negate(_scale));
   }
 
+  /**
+   * @brief Explicit conversion operator for casting to integral types
+   *
+   * @tparam U The integral type that is being explicitly converted to
+   * @return The `fixed_point` number in base 10 (aka human readable format)
+   */
+  template <typename U, typename cuda::std::enable_if_t<std::is_integral<U>::value>* = nullptr>
+  CUDA_HOST_DEVICE_CALLABLE explicit constexpr operator U() const
```

Comment:
Do we need to use `CUDA_HOST_DEVICE_CALLABLE` here, since the function is already `constexpr`?

Reply:
`constexpr` doesn't guarantee compile time.

Reply:
It doesn't need to be guaranteed to evaluate at compile time. `constexpr` functions don't need to be marked `host/device` if compiled with `--relaxed-constexpr` like we do in libcudf.